tests/test_cookiecutter_python_library.py

```diff
@@ -106,22 +106,10 @@ def test_setup_py(options_baked):
     assert "    author_email='cookie@monster.org'," in setup_text
 
 
-def test_upgrade(options_baked):
-    """Make sure the upgrade target works"""
-    run_in_virtualenv('make upgrade')
-
-
-def test_quality(options_baked):
-    """Run quality tests on the given generated output."""
-    for name in all_files():
-        if name.endswith('.py'):
-            sh.pylint(name)
-            sh.pycodestyle(name)
-            sh.pydocstyle(name)
-            sh.isort(name, check_only=True, diff=True)
```

Comment:
What is doing this now? Is it possible to make a change that failed this earlier, and does it still fail now?

Reply:
Now it runs `make test-all`, which calls `tox` and also `make quality`. The latter is here: https://github.com/openedx/edx-cookiecutters/blob/403b620e5a5253714e6e27a7133977ebcd6b7bdd/python-template/%7B%7Bcookiecutter.placeholder_repo_name%7D%7D/Makefile#L50

...except I've now noticed that the doc checks are under the `docs` make target/tox env, which _isn't_ included in the other validation checks. And there were at least three broken things in the base template's docs. I've fixed one so far.

Reply:
I had to go down a bit of a rabbit hole, but all of the templated repo's validation checks should be covered now.